contracts/strategies/CurveYCRVVoter.sol

```diff
@@ -0,0 +1,150 @@
+    function withdraw(IERC20 _asset) external returns (uint256 balance) {
+        require(msg.sender == strategy, "!controller");
+        balance = _asset.balanceOf(address(this));
+        _asset.safeTransfer(strategy, balance);
+    }
+
+    // Withdraw partial funds, normally used with a vault withdrawal
+    function withdraw(uint256 _amount) external {
+        require(msg.sender == strategy, "!controller");
+        uint256 _balance = IERC20(want).balanceOf(address(this));
+        if (_balance < _amount) {
+            _amount = _withdrawSome(_amount.sub(_balance));
+            _amount = _amount.add(_balance);
+        }
+        IERC20(want).safeTransfer(strategy, _amount);
+    }
+
+    // Withdraw all funds, normally used when migrating strategies
+    function withdrawAll() external returns (uint256 balance) {
+        require(msg.sender == strategy, "!controller");
+        _withdrawAll();
+
+        balance = IERC20(want).balanceOf(address(this));
+        IERC20(want).safeTransfer(strategy, balance);
+    }
+
+    function _withdrawAll() internal {
+        Gauge(pool).withdraw(Gauge(pool).balanceOf(address(this)));
+    }
+
+    function createLock(uint256 _value, uint256 _unlockTime) external {
+        require(msg.sender == strategy || msg.sender == governance, "!authorized");
+        IERC20(crv).safeApprove(escrow, 0);
+        IERC20(crv).safeApprove(escrow, _value);
+        VoteEscrow(escrow).create_lock(_value, _unlockTime);
+    }
+
+    function increaseAmount(uint256 _value) external {
+        require(msg.sender == strategy || msg.sender == governance, "!authorized");
+        IERC20(crv).safeApprove(escrow, 0);
+        IERC20(crv).safeApprove(escrow, _value);
+        VoteEscrow(escrow).increase_amount(_value);
+    }
+
+    function release() external {
+        require(msg.sender == strategy || msg.sender == governance, "!authorized");
+        VoteEscrow(escrow).withdraw();
+    }
+
+    function _withdrawSome(uint256 _amount) internal returns (uint256) {
+        Gauge(pool).withdraw(_amount);
+        return _amount;
+    }
+
+    function balanceOfWant() public view returns (uint256) {
+        return IERC20(want).balanceOf(address(this));
+    }
+
+    function balanceOfPool() public view returns (uint256) {
+        return Gauge(pool).balanceOf(address(this));
+    }
+
+    function balanceOf() public view returns (uint256) {
+        return balanceOfWant().add(balanceOfPool());
+    }
+
+    function setGovernance(address _governance) external {
+        require(msg.sender == governance, "!governance");
+        governance = _governance;
+    }
+
+    function execute(
+        address to,
+        uint256 value,
+        bytes calldata data
+    ) external returns (bool, bytes memory) {
+        require(msg.sender == strategy || msg.sender == governance, "!governance");
+        (bool success, bytes memory result) = to.call.value(value)(data);
```

Comment:
This is where the fix should happen right @lbertenasco @banteg

Reply:
Yes :)